File: php/Airport.php
<?php

class Airport {
    
	public string $icao;
    private array $info = [];

    private static $db = null;

	public function __construct(string $icao = 'EGLL') {
		// make icao uppercase
		$this->icao = strtoupper($icao);
        if (self::$db == null) {
            $dbpath = Config::$shared['airport_db_path'];
			self::$db = new PDO("sqlite:$dbpath");
		}
    }

    public static $jsonKeys = ['icao' => 'string'];
    public static $jsonValuesOptionalDefaults = [];

    public function getInfo() {
        if (empty($this->info)) {
		// load airport info from sqlite database in table airports
		// return array with airport info
		//
		//
		// open sqlite database
		$sql = "SELECT * FROM airports WHERE ident = :icao";
		$sth = self::$db->prepare($sql);
		$sth->execute([':icao' => $this->icao]);
		$result = $sth->fetch(PDO::FETCH_ASSOC);
        $this->info = $result;
        }
        return $this->info;
    }

    public function getMapURL() {
        $info = $this->getInfo();
        return "https://www.google.com/maps/place/{$info['latitude_deg']},{$info['longitude_deg']}";
    }

    public function getCity() {
        $info = $this->getInfo();
        return $info['municipality'];
    }
    public function getCountry() {
        $info = $this->getInfo();
        return $info['iso_country'];
    }
    public function getLocation() {
        $info = $this->getInfo();
        if (empty($info['latitude_deg']) || empty($info['longitude_deg'])) {
            return null;
        }
        return [
            'latitude' => $info['latitude_deg'],
            'longitude' => $info['longitude_deg'],
        ];
    }
    public function getIcao() {
        return $this->icao;
    }
    public function getName() {
        $info = $this->getInfo();
        return $info['name'];
    }
    public function fitName(int $maxlen): string{
        $name = $this->getName();
        if (strlen($name) < $maxlen) {
            return $name;
        }else if (strlen($this->getCity()) < $maxlen) {
            return $this->getCity();
        }

        return $name;
    }

}

File: server/app/models/airport.py
"""
Airport domain model.

Matches PHP Airport class structure and JSON serialization.
"""
from typing import Optional
from app.models.base import BaseJsonModel
from app.services.airport_service import AirportService


class Airport(BaseJsonModel):
    """
    Airport model matching PHP Airport class.

    JSON keys match PHP $jsonKeys:
    - icao: str
    - timezone_identifier: str (default "", excluded from JSON)
    """
    icao: str
    timezone_identifier: str = ""

    def get_info(self) -> Optional[dict]:
        """
        Get airport information from AirportService.
        
        Matches PHP: Airport->getInfo()
        """
        return AirportService.get_airport_by_icao(self.icao)

    def get_name(self) -> Optional[str]:
        """
        Get airport name.
        
        Matches PHP: Airport->getName()
        """
        info = self.get_info()
        return info.get('name') if info else None

    def get_city(self) -> Optional[str]:
        """
        Get airport city/municipality.
        
        Matches PHP: Airport->getCity()
        """
        info = self.get_info()
        return info.get('municipality') if info else None

    def get_location(self) -> Optional[dict[str, float]]:
        """
        Get airport location (latitude, longitude).
        
        Matches PHP: Airport->getLocation()
        """
        return AirportService.get_location(self.icao)

    def get_map_url(self) -> Optional[str]:
        """
        Get Google Maps URL for airport.
        
        Matches PHP: Airport->getMapURL()
        """
        return AirportService.get_map_url(self.icao)

    def fit_name(self, maxlen: int) -> str:
        """
        Get airport name or city that fits within maxlen.
        
        Matches PHP: Airport->fitName()
        """
        name = self.get_name() or ""
        city = self.get_city() or ""
        
        if len(name) < maxlen:
            return name
        elif len(city) < maxlen:
            return city
        return name

